### PR TITLE
Always install known molecule drivers

### PR DESCRIPTION
**Pull request by ssbarnea**
This chance makes it possible to use tox-ansible without having to edit tox.ini file at all. By default known drivers will be installed.

Fixes: #52

### CODE REVIEW

**Comment by greg-hellings** [comment]
So is this change going to cause it to install all of the Molecule plugins regardless of what driver it finds? That makes sense, as it allows the user to specify manually "-- --driver foo" to the tox invocation without struggling around deps.

Is there a place other than here that is a sort of meta package that just depends on all of those? Or is this now the canonical location for that list?

**Comment by ssbarnea** [comment]
> So is this change going to cause it to install all of the Molecule plugins regardless of what driver it finds? That makes sense, as it allows the user to specify manually "-- --driver foo" to the tox invocation without struggling around deps.
> 
> Is there a place other than here that is a sort of meta package that just depends on all of those? Or is this now the canonical location for that list?

We do not have a meta package but we do happen to have https://github.com/ansible-community/toolset/blob/main/requirements.txt which is very close to that: that is a container that contains most plugins, or at least those that we were able to pack together.

It can become tricky and expensive to install all plugins, so for the moment I think we need to limit to the popular ones and later find a better way to identify which ones are needed for an entire repository.

File: src/tox_ansible/tox_molecule_case.py [hunk 1]
```diff
@@ -67,12 +69,14 @@ def get_dependencies(self):
...
-        ):
-            dependencies.extend(DRIVER_DEPENDENCIES[self.scenario.driver])
+        # Drivers can have their own dependencies but we preinstall of known
+        # ones as the current venv may be shared among multiple scenarios
+        # If this proves to become heavy, we should look into finding a way
+        # to optimize it by allowing user to enable a slim mode.
+        for _, deps in DRIVER_DEPENDENCIES.items():
+            for dep in deps:
+                if dep not in dependencies:
+                    dependencies.append(dep)
```
Comment (greg-hellings) [inline src/tox_ansible/tox_molecule_case.py:79]: If this is flattening all of the dependencies into one, we can remove the redundancy between podman and docker that I put into the original deps list up at lines 12-13.

Comment (ssbarnea) [reply src/tox_ansible/tox_molecule_case.py:79]: Yeah. I know but i will postpone it for now. I have the impression that we will be revisiting the logic soon anyway.

For the moment I want to focus on bringing this plugin to production for tripleo repos. Once users will see how easy is to use, I expect to start seeing features requests being raised.

**Pull request merged**